File: app/src/main/java/com/wildlifeoftianjin/ui/activity/EditRecordActivity.java
package com.wildlifeoftianjin.ui.activity;

import android.os.Bundle;
import android.support.v4.app.FragmentTransaction;
import android.view.View;

import com.wildlifeoftianjin.R;
import com.wildlifeoftianjin.data.NetResult;
import com.wildlifeoftianjin.model.Record;
import com.wildlifeoftianjin.network.Request.UploadRecordRequest;
import com.wildlifeoftianjin.network.Request.UploadRequest;
import com.wildlifeoftianjin.ui.fragment.EditRecordFragment;

/**
 * Created by presisco on 2017/5/18.
 */

public class EditRecordActivity extends NetworkActivity
        implements UploadRequest.UploadResponse {
    public static final String KEY_CREATURE_NAME = "name";

    private EditRecordFragment mEditRecordFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_record);

        mEditRecordFragment = new EditRecordFragment();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.recordFragment, mEditRecordFragment);
        transaction.commit();

        String creature_name = getIntent().getStringExtra(KEY_CREATURE_NAME);
        mEditRecordFragment.setCreatureName(creature_name);
    }

    public void onUpload(View v) {
        Record record = mEditRecordFragment.getRecord();
        getRequestQueue().add(new UploadRecordRequest(record, this, this));
    }

    @Override
    public void onResponse(NetResult result) {

    }
}

File: app/src/main/java/com/wildlifeoftianjin/ui/activity/EditRecordActivity.kt
package com.wildlifeoftianjin.ui.activity

import android.os.Bundle
import android.view.View
import android.widget.Toast

import com.android.volley.Response
import com.wildlifeoftianjin.R
import com.wildlifeoftianjin.network.Constants
import com.wildlifeoftianjin.network.Request.PostFormRequest
import com.wildlifeoftianjin.ui.fragment.EditRecordFragment

/**
 * Created by presisco on 2017/5/18.
 */

class EditRecordActivity : NetworkActivity(), Response.ErrorListener {

    private var mEditRecordFragment: EditRecordFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_record)

        mEditRecordFragment = EditRecordFragment()

        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.record_fragment, mEditRecordFragment)
        transaction.commit()

        val creature_name = intent.getStringExtra(KEY_CREATURE_NAME)
        val creature_id = intent.getStringExtra(KEY_CREATURE_ID)
        mEditRecordFragment!!.setCreature(creature_name, creature_id)
    }

    fun onUpload(v: View): Unit {
        val record = mEditRecordFragment!!.record
        requestQueue.add(PostFormRequest(Constants.PATH_UPLOAD_RECORD, record, onResponse, this))
        showLoadingIndicator()
    }

    val onResponse = {
        result: Map<String, String> ->
        var toast_content = ""
        if (result["result"] === "1") {
            toast_content = "upload succeed"
        } else {
            toast_content = "upload failed"
        }
        Toast.makeText(this, toast_content, Toast.LENGTH_SHORT).show()
        hideLoadingIndicator()
    }

    companion object {
        val KEY_CREATURE_NAME = "name"
        val KEY_CREATURE_ID = "id"
    }
}
